skorch/tests/test_classifier.py

```diff
@@ -72,6 +72,13 @@ def test_score(self, net_fit, data):
         accuracy = net_fit.score(X, y)
         assert 0. <= accuracy <= 1.
 
+    def test_check_data_accepts_skorch_dataset(self, net, data):
+        from skorch.dataset import Dataset
+
+        X, y = data
+        dataset = Dataset(X, y)
+        assert net.check_data(dataset, y=None) is None
```

Comment:
```suggestion
    def test_fit_with_dataset_and_y_none(self, net_cls, module_cls, data):
        from skorch.dataset import Dataset

        # deactivate train split since it requires y
        net = net_cls(module_cls, train_split=False, max_epochs=1)
        X, y = data
        dataset = Dataset(X, y)
        assert net.fit(dataset, y=None)
```